src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -666,23 +643,23 @@ def prepare_create(self, daemon_spec: CephadmDaemonDeploySpec) -> CephadmDaemonD
         daemon_spec.final_config, daemon_spec.deps = self.generate_config(daemon_spec)
         return daemon_spec
 
+    def get_node_exporter_certificates(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[str, str]:
+        node_ip = self.mgr.inventory.get_addr(daemon_spec.host)
+        host_fqdn = self._inventory_get_fqdn(daemon_spec.host)
+        cert, key = self.mgr.cert_mgr.generate_cert(host_fqdn, node_ip)
+        return cert, key
+
     def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[str, Any], List[str]]:
         assert self.TYPE == daemon_spec.daemon_type
         deps = [f'secure_monitoring_stack:{self.mgr.secure_monitoring_stack}']
         if self.mgr.secure_monitoring_stack:
-            node_ip = self.mgr.inventory.get_addr(daemon_spec.host)
-            host_fqdn = self._inventory_get_fqdn(daemon_spec.host)
-            cert = self.mgr.cert_key_store.get_cert('node_exporter_cert', host=daemon_spec.host)
-            key = self.mgr.cert_key_store.get_key('node_exporter_key', host=daemon_spec.host)
-            if not (cert and key):
-                cert, key = self.mgr.http_server.service_discovery.ssl_certs.generate_cert(
-                    host_fqdn, node_ip)
-                self.mgr.cert_key_store.save_cert('node_exporter_cert', cert, host=daemon_spec.host)
-                self.mgr.cert_key_store.save_key('node_exporter_key', key, host=daemon_spec.host)
+            cert, key = self.get_node_exporter_certificates(daemon_spec)
+            mgmt_gw_enabled = len(self.mgr.cache.get_daemons_by_service('mgmt-gateway')) > 0
```

Comment:
Just a general point about this way of telling the mgmt gw is setup. If we are using this do make configuration decisions, we may need the same thing to be added to the deps for the daemon so we can update its configuration if this changes

Reply:
`mgmt-gateway` is a dep for all all the monitoring services, isn't that enough?